lib/mayaUsd/render/vp2RenderDelegate/mesh.cpp

```diff
@@ -136,10 +136,20 @@ namespace {
             if (requiresUnsharedVertices) {
                 const VtIntArray& faceVertexIndices = topology.GetFaceVertexIndices();
                 if (numVertices == faceVertexIndices.size()) {
+                    unsigned int dataSize = primvarData.size();
                     for (size_t v = 0; v < numVertices; v++) {
                         SRC_TYPE* pointer = reinterpret_cast<SRC_TYPE*>(
                             reinterpret_cast<float*>(&vertexBuffer[v]) + channelOffset);
-                        *pointer = primvarData[faceVertexIndices[v]];
+                        unsigned int index = faceVertexIndices[v];
+                        if (index < dataSize) {
+                            *pointer = primvarData[faceVertexIndices[v]];
+                        } else {
+                            TF_DEBUG(HDVP2_DEBUG_MESH).Msg("Invalid face vertex index '%s': "
+                                                    "primvar %s has %zu elements, while its topology "
+                                                    "references only upto element index %zu.\n",
+                                                    rprimId.asChar(), primvarName.GetText(),
+                                                    primvarData.size(), numVertices);                            
```

Comment:
primvarData.size() == dataSize...as for debugging...wouldn't knowing `index` be more useful than `numVertices` ?

Reply:
You're talking about the print statement printing out index?  Maybe I suppose, although with this kind of thing the actual # might not help you at all.  The fact that something was out of bounds + the name of the object is what you probably need and not the actual number.

Reply:
Agreed. Then how about we simply remove # from the message?

Reply:
Sorry what do you mean?